server/src/main/java/org/elasticsearch/search/aggregations/metrics/MaxAggregator.java

```diff
@@ -68,8 +88,23 @@ public ScoreMode scoreMode() {
     public LeafBucketCollector getLeafCollector(LeafReaderContext ctx,
             final LeafBucketCollector sub) throws IOException {
         if (valuesSource == null) {
-            return LeafBucketCollector.NO_OP_COLLECTOR;
-    }
+            if (parent != null) {
+                return LeafBucketCollector.NO_OP_COLLECTOR;
+            } else {
+                // we have no parent and the values source is empty so we can skip collecting hits.
+                throw new CollectionTerminatedException();
+            }
+        }
+        if (pointConverter != null) {
+            Number segMax = findLeafMaxValue(ctx.reader(), pointField, pointConverter);
+            if (segMax != null) {
+                double max = maxes.get(0);
```

Comment:
Could you add a comment explaining why its ok that we always use the bucket ordinal `0` here?